### PR TITLE
Use new validated commit

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr

- Uses `validated_commit_v2` everywhere
- Renames `validated_commit_v2` to `validated_commit`

## More info

- The new validated_commit struct includes a diff of all metadata changes. This requires a change to how we evaluate permissions, which previously had to figure out the diff as part of evaluation. This should be easier to test and work with.
- Using the new validated commit requires a bunch of functions that were previously synchronous to be async. We are using the `await_helper` to wrap these synchronous functions in async functions. We are going to have to watch this carefully for deadlocks. Alternatively, maybe we want to create an `async_transaction` function on our `DbConnection` that works similarly to the existing `transaction` function but allows for an async callback.

### CODE REVIEW

**Comment by neekolas** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/740?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#740** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/740?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#739** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/739?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#718** <a href="https://app.graphite.dev/github/pr/xmtp/libxmtp/718?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @neekolas and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: xmtp_mls/src/await_helper.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+use futures::Future;
+use tokio::{runtime::Handle, task};
+
+pub fn await_helper<F: Future + Send>(future: F) -> F::Output
+where
+    <F as Future>::Output: Send + 'static,
+{
+    task::block_in_place(move || Handle::current().block_on(future))
+}
```
Comment (37ng) [inline xmtp_mls/src/await_helper.rs:9]: It'd nice if we can start putting helpers/utils into a central place e.g. `xmtp_mls/src/utils/...`

iirc you mentioned runtime issues this may have, wondering if this is for temp usage or considered valid.

File: xmtp_mls/src/groups/sync.rs [hunk 2]
```diff
@@ -457,7 +480,7 @@ impl MlsGroup {
             EntityKind::Group,
             msgv1.id,
             |provider| -> Result<(), MessageProcessingError> {
-                self.process_message(openmls_group, provider, msgv1, true)?;
+                await_helper(self.process_message(client, openmls_group, provider, msgv1, true))?;
```
Comment (37ng) [inline xmtp_mls/src/groups/sync.rs:483]: looks like we don't need `await_helper` here given that `process_message` is async?

**Review by 37ng — APPROVED** [review]

File: xmtp_mls/src/await_helper.rs (hunk 1 above)
Comment (neekolas) [reply xmtp_mls/src/await_helper.rs:9]: Hopefully it's for temp usage. @insipx is going to make a `transaction_async` method for the store that will make this obsolete

File: xmtp_mls/src/groups/sync.rs (hunk 2 above)
Comment (neekolas) [reply xmtp_mls/src/groups/sync.rs:483]: `process_message` is async, but `client.process_for_id` requires the callback to be synchronous because it runs in a transaction.

This should all go away once we have asynchronous transaction support

**Comment by neekolas** [comment]
I'm going to merge this now so that @cameronvoell can keep working on permissions on top of `identity-release`. But feel free to keep commenting or adding notes and I can fix in a subsequent PR

File: xmtp_mls/src/groups/sync.rs (hunk 2 above)
Comment (37ng) [reply xmtp_mls/src/groups/sync.rs:483]: nice
